From 746224005321f24e78cfbbc1b59b5e1378199612 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@fedoraproject.org>
Date: Mon, 19 Jan 2026 12:24:39 -0500
Subject: [PATCH] client/systray: Short circuit installSystemd() if a global
 unit exists

When a packaged version of the unit exists, we do not want one to
be created for us, instead bail early and tell the user to enable
the unit themselves.
---
 client/systray/startup-creator.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/client/systray/startup-creator.go b/client/systray/startup-creator.go
index cb354856d..0beb490a5 100644
--- a/client/systray/startup-creator.go
+++ b/client/systray/startup-creator.go
@@ -30,6 +30,14 @@ func InstallStartupScript(initSystem string) error {
 }
 
 func installSystemd() error {
+	// Check if a global tailscale-systray.service unit file exists
+	// If it exists, bail and just print out steps to enable it.
+	_, err := os.Stat("/usr/lib/systemd/user/tailscale-systray.service")
+	if err == nil {
+		fmt.Println("To enable and start the service, run:")
+		fmt.Println("  systemctl --user enable --now tailscale-systray")
+		return nil
+	}
 	// Find the path to tailscale, just in case it's not where the example file
 	// has it placed, and replace that before writing the file.
 	tailscaleBin, err := exec.LookPath("tailscale")
-- 
2.52.0

